base9/Settings.cpp

```diff
@@ -509,6 +511,24 @@ template<typename T> T Settings::getOrDie (Node & n, string && f)
     }
 }
 
+template<typename T> T Settings::getOrRequest (Node & n, string && f)
+{
+    if (n[f])
+    {
+        return n[f].as<T> ();
+    }
+    else
+    {
+        string input = "";
+        cout << "Please enter your desired setting for field: '" + f + "':\n";
+        getline(cin, input);
+        cout << "Field '" + f + "' set to: " << input << endl;
+        YAML::Node node;
```

Comment:
I wish there were a member function such that you could do e.g.,

```cpp
    return istringstream(input).extract();
```

It appears the inelegant temporary variable will have to suffice.